### PR TITLE
nixos/pipewire: only enable media-session if pipewire is enabled

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/desktops/pipewire/pipewire-media-session.nix [hunk 1]
```diff
@@ -37,7 +37,8 @@ in {
     services.pipewire.media-session = {
       enable = mkOption {
         type = types.bool;
-        default = true;
+        default = config.pipewire.enable;
```
Comment (minijackson) [inline nixos/modules/services/desktops/pipewire/pipewire-media-session.nix:40]: Shouldn't it be `config.services.pipewire.enable`?

Comment (Mic92) [reply nixos/modules/services/desktops/pipewire/pipewire-media-session.nix]: Yeah. I messed up cherry-picking. I tested with a different commit.

**Review by jansol — APPROVED** [review]
Works with pipewire enabled. Didn't test whether it still pulls in pipewire when disabled.

**Review by samueldr — APPROVED** [review]
Approving since I verified it fixes the issue about it being enabled by default in an unconfigured system. Have not validated the behaviour is okay otherwise.

**Pull request merged**